Dfc.Api.DiscoverSkillsAndCareers/Functions/NewAssessmentFunctions.cs

```diff
@@ -104,11 +113,63 @@ public async Task<IActionResult> CreateNewShortAssessment([HttpTrigger(Authoriza
         {
             request.LogRequestHeaders(logService);
 
-            return responseWithCorrelation.ResponseWithCorrelationId(HttpStatusCode.Created);
+            var correlationId = Guid.Parse(correlationIdProvider.CorrelationId);
+            logService.LogMessage($"CorrelationId: {correlationId} - Creating a new assessment", SeverityLevel.Information);
+
+            using var streamReader = new StreamReader(request.Body);
+            var requestBody = streamReader.ReadToEnd();
+            var dfcUserSession = JsonConvert.DeserializeObject<DfcUserSession>(requestBody);
+
+            if (dfcUserSession is null)
+            {
+                logService.LogMessage($"CorrelationId: {correlationId} - Request Body is empty", SeverityLevel.Warning);
+                return responseWithCorrelation.ResponseWithCorrelationId(HttpStatusCode.BadRequest, correlationId);
+            }
+
+            var isValid = sessionClient.ValidateUserSession(dfcUserSession);
+            if (!isValid)
+            {
+                logService.LogMessage($"CorrelationId: {correlationId} - Session with Id {dfcUserSession.SessionId} is not valid", SeverityLevel.Warning);
+                return responseWithCorrelation.ResponseWithCorrelationId(HttpStatusCode.BadRequest, correlationId);
+            }
+
+            var currentQuestionSetInfo = await questionSetRepository.GetCurrentQuestionSet(SkillsAssessmentType).ConfigureAwait(false);
+            if (currentQuestionSetInfo == null)
+            {
+                logService.LogMessage($"CorrelationId: {correlationId} - Unable to load latest question set {SkillsAssessmentType}", SeverityLevel.Warning);
+                return responseWithCorrelation.ResponseWithCorrelationId(HttpStatusCode.NoContent, correlationId);
+            }
+
+            var existingUserSession = await userSessionRepository.GetAsync(f => f.UserSessionId == dfcUserSession.SessionId).ConfigureAwait(false);
+            if (existingUserSession != null)
+            {
+                logService.LogMessage($"Unable to create User Session with id {dfcUserSession.SessionId}. This record already exists", SeverityLevel.Information);
+                return responseWithCorrelation.ResponseWithCorrelationId(HttpStatusCode.AlreadyReported, correlationId);
+            }
+
+            await CreateUserSession(currentQuestionSetInfo, dfcUserSession).ConfigureAwait(false);
+
+            return responseWithCorrelation.ResponseWithCorrelationId(HttpStatusCode.Created, correlationId);
+        }
+
+        private async Task CreateUserSession(QuestionSet currentQuestionSetInfo, DfcUserSession dfcUserSession)
+        {
+            var userSession = new UserSession
+            {
+                UserSessionId = dfcUserSession.SessionId,
+                Salt = dfcUserSession.Salt,
+                StartedDt = dfcUserSession.CreatedDate,
+                LanguageCode = "en",
+                PartitionKey = dfcUserSession.PartitionKey,
```

Comment:
Shouldn't be using the same partition key, as this may not be our partition prefix. Use the nuget package to create partition key.

Reply:
Fetched the partitionkey from nuget